pkg/runtime/framework/plugins/volcano/volcano_test.go

```diff
@@ -0,0 +1,399 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	batchv1ac "k8s.io/client-go/applyconfigurations/batch/v1"
+	corev1ac "k8s.io/client-go/applyconfigurations/core/v1"
+	metav1ac "k8s.io/client-go/applyconfigurations/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client/interceptor"
+	jobsetv1alpha2 "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+	jobsetv1alpha2ac "sigs.k8s.io/jobset/client-go/applyconfiguration/jobset/v1alpha2"
+	volcanov1beta1 "volcano.sh/apis/pkg/apis/scheduling/v1beta1"
+	"volcano.sh/apis/pkg/client/applyconfiguration/scheduling/v1beta1"
+	volcanov1beta1ac "volcano.sh/apis/pkg/client/applyconfiguration/scheduling/v1beta1"
+
+	trainer "github.com/kubeflow/trainer/v2/pkg/apis/trainer/v1alpha1"
+	"github.com/kubeflow/trainer/v2/pkg/apply"
+	"github.com/kubeflow/trainer/v2/pkg/runtime"
+)
+
+func TestVolcano(t *testing.T) {
+	scheme := apiruntime.NewScheme()
+	if err := trainer.AddToScheme(scheme); err != nil {
+		t.Fatalf("failed to add trainer scheme: %v", err)
+	}
+	if err := volcanov1beta1.AddToScheme(scheme); err != nil {
+		t.Fatalf("failed to add volcano scheme: %v", err)
+	}
+
+	ctx := context.Background()
+	// Test Name()
+	t.Run("Name", func(t *testing.T) {
+		v := &Volcano{}
+		expectedName := "Volcano"
+		if v.Name() != expectedName {
+			t.Errorf("expected name %s, got %s", expectedName, v.Name())
+		}
+	})
+
+	// Test ReconcilerBuilders()
+	t.Run("ReconcilerBuilders", func(t *testing.T) {
+		gvk := volcanov1beta1.SchemeGroupVersion.WithKind("PodGroup")
+		mapper := meta.NewDefaultRESTMapper([]schema.GroupVersion{volcanov1beta1.SchemeGroupVersion})
+		mapper.Add(gvk, meta.RESTScopeNamespace)
+		v := &Volcano{
+			scheme:     scheme,
+			client:     fake.NewClientBuilder().WithScheme(scheme).Build(),
+			restMapper: mapper,
+		}
+		builders := v.ReconcilerBuilders()
+		if diff := cmp.Diff(3, len(builders)); diff != "" {
+			t.Errorf("unexpected builder count (-want +got):\n%s", diff)
+		}
+	})
+
+	// Test Build()
+	mockGetError := errors.New("mock get error")
+
+	jobSetSpecApply, err := apply.FromTypedObjWithFields[jobsetv1alpha2ac.JobSetSpecApplyConfiguration](&jobsetv1alpha2.JobSet{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: jobsetv1alpha2.GroupVersion.String(),
+			Kind:       "JobSet",
+		},
+		Spec: jobsetv1alpha2.JobSetSpec{
+			ReplicatedJobs: []jobsetv1alpha2.ReplicatedJob{
+				{
+					Name: "launcher",
+					Template: batchv1.JobTemplateSpec{
+						Spec: batchv1.JobSpec{
+							Template: corev1.PodTemplateSpec{
+								Spec: corev1.PodSpec{
+									PriorityClassName: "high-priority",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}, "spec")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	baseInfo := func() *runtime.Info {
```

Comment:
```suggestion
	baseInfo :=  &runtime.Info {
```

I don't think that we should have anonymous function.